client/src/OrderForm.jsx

```diff
@@ -0,0 +1,167 @@
+    this.quantityOptions = makeQuantityOptions(this.props.data.quantity).slice(1);
+
+    this.state = {
+      singlePrice: lowestPrice,
+      totalPrice: lowestPrice,
+      quantity: 1,
+      variation: '',
+      pleaseSelectShown: false, // e.g. "Please select a size"
+    };
+
+    this.handleVariationSelect = this.handleVariationSelect.bind(this);
+    this.handleQuantitySelect = this.handleQuantitySelect.bind(this);
+    this.handleBuyNowClick = this.handleBuyNowClick.bind(this);
+  }
+
+  handleVariationSelect(event) {
+    // Passing a tuple in as a select option's value converts it to a string
+    // separated by commas. This would potentially create problems
+    // if the description had commas in it.
+    const [description, price] = event.target.value.split(',');
+    this.setState({
+      singlePrice: price,
+      variation: description,
+      pleaseSelectShown: false,
+    });
+  }
+
+  handleQuantitySelect(event) {
+    const quantity = +event.target.value;
+    this.setState({
+      quantity,
+      totalPrice: this.state.singlePrice * quantity,
+    });
+  }
+
+  handleBuyNowClick(event) {
+    if (!this.state.variation) {
+      this.setState({ pleaseSelectShown: true });
+    }
+    // More stuff will go here when I add modals
+  }
+
+  renderPleaseSelect(variationType) {
+    return this.state.pleaseSelectShown ?
+      <div className="please-select">
+        Please select a {variationType.toLowerCase()}
+      </div> :
+      null;
+  }
+
+  render() {
+    return (
+      <div className="main-item" id="order-form">
+        <h4 id="title">{this.props.data.title}</h4>
+
+        <div id="price-and-question">
+          <span id="price">${this.state.totalPrice}</span>
+          <span id="q-span"><button id="q-button">Ask a question</button></span>
+        </div>
+
+        <div id="variations">
+          {this.props.data.variationTypes.map(type => (
+            <div key={type} className="variation">
+              <div className="variation-name">{type}</div>
+
+              <select onChange={this.handleVariationSelect}>
+
+                {this.props.data.variations[type]
+                  .map((variationTuple) => {
+                    const [description, price] = variationTuple;
+
+                    return (
+                      <option key={description} value={variationTuple}>
+                        {`${description} ($${price})`}
+                      </option>
+                    );
+                  })
+                }
+
+              </select>
+
+              {this.renderPleaseSelect(type)}
+            </div>
+          ))}
+        </div>
+
+        <div id="quantity">
+          <div>Quantity</div>
+          <select onChange={this.handleQuantitySelect}>
+            {this.quantityOptions}
+          </select>
+        </div>
+
+        <div id="buy-now">
+          <button onClick={this.handleBuyNowClick}>
+          Buy it now {'>'}
```

Comment:
I should have left a comment - it just renders the character > on the page (because the Etsy component has "Buy it now >"). I had to make it a string in code brackets because JSX doesn't like characters that could be HTML but aren't formatted properly for it.